dashboard/src/utils/constructDate.js

```diff
@@ -0,0 +1,3 @@
+export const constructNewDate = (date) => {
+  return new Date(`${date.split(":")[0]}T00:00:00`);
+};
```

Comment:
In this context, `date` is a _string_, right?  Since in the surrounding contexts you've used `date` to pass a `Date` object, it would be good if the parameter here we're named something more like `date_str` or `string_date`.

Do we really need/want to convert the input date-_time_ to a date-at-midnight?  That really shouldn't be necessary, right?  That is, whether it has a time or not shouldn't affect the selection process later (the time-at-midnight is really only a concern for the values that come from Patternfly).  But, it does make things more awkward here, and it exposes us to timezone problems which we could otherwise finesse.

That is, if the incoming date-time string includes the timezone indicator (e.g., if it's in the form `YYYY-MM-DDThh:mm:ss.fffZ` with the trailing `Z`), then we should be able to convert it directly into a `Date` object without manipulation, and it should reflect the correct value without having to account for local time.

Reply:
True; we shouldn't arbitrarily change the time-of-day for any dates except when adding time-of-day to the *dates* provided by the date picker (so that the range is from the beginning of one day to the end of the other). Any timestamp returned by the server is either inside or outside that range when timezone adjusted.

We currently return the `YYYY-MM-DD:hh:mm` format from the server, implicitly in UTC. We *should* change that to ISO 8601 `YYYY-MM-DDThh:mm:ss[Z|(+|-)nn` format for consistency and clarity.

Reply:
And, in the meantime, @shubham-html-css-js should add the `Z` to the server timestamp (instead of the butchery here).